Code/src/main/java/nl/utwente/viskell/ui/CustomUIPane.java

```diff
@@ -66,10 +67,8 @@ public CustomUIPane(HaskellCatalog catalog) {
         this.addEventHandler(KeyEvent.KEY_PRESSED, this::handleKey);
 
         /* Run the Inspector window. */
-        Platform.runLater(() -> {
-            inspector = new InspectorWindow(this);
-            inspector.blockProperty().bind(this.selectedBlock);
-        });
+        Platform.runLater(() -> inspector = new InspectorWindow(this));
+        Platform.runLater(() -> preferences = new PreferencesWindow(this));
```

Comment:
I'm not sure. Technically, I don't believe it's guaranteed that `showPreferences` is called on the JavaFX Application Thread, but in our case it might be. I'll check this out.